### PR TITLE
Top down biomarker

### PR DESCRIPTION
**Pull request by trishorts**
may have a couple mini rounds to smooth our wrinkles w/ the search in mm

This is now designed to work hand in glove w/ MM. Default in MM will be to retain methionine when biomarker search is selected. Than the M will get nibbled away at the start by proteolysis and it will get covered w/ and w/o M.

### CODE REVIEW

File: mzLib.sln [hunk 1]
```diff
@@ -0,0 +1,140 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
[the remark this answers is not in the document]
Comment (acesnik) [reply mzLib.sln:2]: I think you can just delete this one

File: mzLib/Test/TestRawFileReader.cs [hunk 2]
```diff
@@ -19,9 +17,6 @@ public sealed class TestRawFileReader
         [TestCase("testFileWMS2.raw", "a.mzML", "aa.mzML")]
         [TestCase("small.raw", "a.mzML", "aa.mzML")]
         [TestCase("05-13-16_cali_MS_60K-res_MS.raw", "a.mzML", "aa.mzML")]
-        /// <summary>
-        /// Tests LoadAllStaticData for ThermoRawFileReader
```
Comment (acesnik) [inline mzLib/Test/TestRawFileReader.cs:23]: Codemaid deletes method descriptions below the test cases. Could you add these back above `[Test]`?

File: mzLib/Test/TestRawFileReader.cs [hunk 3]
```diff
@@ -39,9 +34,6 @@ public static void TestLoadAllStaticDataRawFileReader(string infile, string outf
         }
 
         [Test]
-        /// <summary>
-        /// Tests the dynamic connection for ThermoRawFileReader
```
Comment (acesnik) [inline mzLib/Test/TestRawFileReader.cs:43]: here too, and more below

File: mzLib/UsefulProteomicsDatabases/ProteinDbLoader.cs [hunk 4]
```diff
@@ -273,17 +273,13 @@ public static List<Modification> GetPtmListFromProteinXml(string proteinDbLocati
...
+                            isContaminant: isContaminant, databaseFilePath: proteinDbLocation, addBiomarkers: addBiomarkers);
                         if (protein.Length == 0)
                         {
                             errors.Add("Line" + line + ", Protein Length of 0: " + protein.Name + " was skipped from database: " + proteinDbLocation);
                         }
                         else
                         {
-                            if (addBiomarkers)
-                            {
-                                protein.AddBiomarkers();
```
Comment (acesnik) [inline mzLib/UsefulProteomicsDatabases/ProteinDbLoader.cs:285]: So this is now just called from `Protein`, it looks like?

Comment (trishorts) [reply mzLib/UsefulProteomicsDatabases/ProteinDbLoader.cs:285]: yes

File: mzLib/Test/TestRawFileReader.cs (hunk 3 above)
Comment (trishorts) [reply mzLib/Test/TestRawFileReader.cs:43]: done

File: mzLib/Test/TestRawFileReader.cs (hunk 2 above)
Comment (trishorts) [reply mzLib/Test/TestRawFileReader.cs:23]: i didnt know that. i'm putting them back

File: mzLib/Test/TestProteinDatabase.cs [hunk 5]
```diff
@@ -7,100 +7,10 @@
 
 namespace Test
 {
-    [TestFixture]
     [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    [TestFixture]
     public sealed class TestProteinDatabase
     {
-        [Test]
-        public static void MakeAnewProteinWithAndWithoutBiomarkers()
```
[the remark this answers is not in the document]
Comment (acesnik) [reply mzLib/Test/TestProteinDatabase.cs]: 👍🏼

File: mzLib/Test/TestRawFileReader.cs (hunk 2 above)
Comment (acesnik) [reply mzLib/Test/TestRawFileReader.cs:23]: Yeah, it's not the greatest behavior on Codemaid's part in my opinion!

**Review by acesnik — APPROVED** [review]
This looks good. 

If you want to avoid iterating with mzLib PRs with testing in MM, you can make local packages with `nuget pack` in the file with the `.nuspec` file, and then you can create a local feed for mzLib:
image

**Comment by acesnik** [comment]
Approved

**Review by rmmiller22 — APPROVED** [review]

**Pull request merged**